src/logger.ts

```diff
@@ -16,261 +16,252 @@ import {
 import LogMonitor from './logging/monitor';
 
 import * as globalEventBus from './event-bus';
-import log from './lib/supervisor-console';
-
-interface LoggerSetupOptions {
-	apiEndpoint: config.ConfigType<'apiEndpoint'>;
-	uuid: config.ConfigType<'uuid'>;
-	deviceApiKey: config.ConfigType<'deviceApiKey'>;
-	unmanaged: config.ConfigType<'unmanaged'>;
-	localMode: config.ConfigType<'localMode'>;
-
-	enableLogs: boolean;
-}
+import superConsole from './lib/supervisor-console';
 
 type LogEventObject = Dictionary<any> | null;
 
-export class Logger {
-	private backend: LogBackend | null = null;
-	private balenaBackend: BalenaLogBackend | null = null;
-	private localBackend: LocalLogBackend | null = null;
+// export class Logger {
+let backend: LogBackend | null = null;
+let balenaBackend: BalenaLogBackend | null = null;
+let localBackend: LocalLogBackend | null = null;
 
-	private containerLogs: { [containerId: string]: ContainerLogs } = {};
-	private logMonitor: LogMonitor;
-
-	public constructor() {
-		this.backend = null;
-		this.logMonitor = new LogMonitor();
-	}
+const containerLogs: { [containerId: string]: ContainerLogs } = {};
+const logMonitor = new LogMonitor();
 
-	public init({
+export const initialized = (async () => {
+	await config.initialized;
+	const {
 		apiEndpoint,
 		uuid,
 		deviceApiKey,
 		unmanaged,
-		enableLogs,
+		loggingEnabled,
 		localMode,
-	}: LoggerSetupOptions) {
-		this.balenaBackend = new BalenaLogBackend(apiEndpoint, uuid, deviceApiKey);
-		this.localBackend = new LocalLogBackend();
-
-		this.backend = localMode ? this.localBackend : this.balenaBackend;
-
-		this.backend.unmanaged = unmanaged;
-		this.backend.publishEnabled = enableLogs;
-
-		// Only setup a config listener if we have to
-		if (!this.balenaBackend.isIntialised()) {
-			globalEventBus.getInstance().once('deviceProvisioned', async () => {
-				const conf = await config.getMany([
-					'uuid',
-					'apiEndpoint',
-					'deviceApiKey',
-				]);
-
-				// We use Boolean here, as deviceApiKey when unset
-				// is '' for legacy reasons. Once we're totally
-				// typescript, we can make it have a default value
-				// of undefined.
-				if (_.every(conf, Boolean)) {
-					// Everything is set, provide the values to the
-					// balenaBackend, and remove our listener
-					this.balenaBackend!.assignFields(
-						conf.apiEndpoint,
-						conf.uuid!,
-						conf.deviceApiKey,
-					);
-				}
-			});
-		}
+	} = await config.getMany([
+		'apiEndpoint',
+		'uuid',
+		'deviceApiKey',
+		'unmanaged',
+		'loggingEnabled',
+		'localMode',
+	]);
+
+	balenaBackend = new BalenaLogBackend(apiEndpoint, uuid, deviceApiKey);
+	localBackend = new LocalLogBackend();
+	backend = localMode ? localBackend : balenaBackend;
+	backend.unmanaged = unmanaged;
+	backend.publishEnabled = loggingEnabled;
+
+	if (!balenaBackend.isIntialised()) {
```

Comment:
```suggestion
	if (!balenaBackend.isInitialised()) {
```

Reply:
:S why wasn't this picked up by typescript. That's concerning, I'll have to check it out.

Reply:
Ah, consistent typos :)